upgrade/helpers/tools.py

```diff
@@ -95,6 +96,45 @@ def host_pings(host, timeout=15):
             time.sleep(5)
 
 
+def host_ssh_check(host, timeout=15):
+    """This ensures the given IP/hostname pings succesfully.
+
+    :param host: A string. The IP or hostname of host.
+    :param int timeout: The polling timeout in minutes.
+
+    """
+    get_ip = subprocess.Popen(
+        'ping -c1 {0}; echo $?'.format(host),
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        shell=True
+    )
+    ip_command = get_ip.communicate()[0]
+    if int(ip_command.split()[-1]) == 0:
+        ip = ip_command[ip_command.find("(")+1:ip_command.find(")")]
+    else:
+        logger.warning('Ip address for host {0} not found'.format(host))
+        sys.exit(1)
+    timeup = time.time() + int(timeout) * 60
```

Comment:
Better to add new function in upgrade.tools for code henceforth in this function that checks the 'nc' (route to host) output.
Just call that function here